examples/naive/aarch64/intt_dilithium_123_45678.s

```diff
@@ -0,0 +1,533 @@
+
+// Needed to provide ASM_LOAD directive
+#include <hal_env.h>
+
+// NOTE
+// We use a lot of trivial macros to simplify the parsing burden for Slothy
+// The macros are not unfolded by Slothy and thus interpreted as instructions,
+// which are easier to parse due to e.g. the lack of size specifiers and simpler
+// syntax for pre and post increment for loads and stores.
+//
+// Eventually, NeLight should include a proper parser for AArch64,
+// but for initial investigations, the below is enough.
+xtmp0 .req x10
+xtmp1 .req x11
+
+.macro ldr_vo vec, base, offset
+       ldr qform_\vec, [\base, #\offset]
+.endm
+
+.macro ldr_vi vec, base, inc
+        ldr qform_\vec, [\base], #\inc
+.endm
+
+.macro str_vo vec, base, offset
+        str qform_\vec, [\base, #\offset]
+.endm
+.macro str_vi vec, base, inc
+        str qform_\vec, [\base], #\inc
+.endm
+.macro vqrdmulh d,a,b
+        sqrdmulh \d\().4s, \a\().4s, \b\().4s
+.endm
+.macro vmls d,a,b
+        mls \d\().4s, \a\().4s, \b\().4s
+.endm
+.macro vqrdmulhq d,a,b,i
+        sqrdmulh \d\().4s, \a\().4s, \b\().s[\i]
+.endm
+.macro vqdmulhq d,a,b,i
+        sqdmulh \d\().4s, \a\().4s, \b\().4s[\i]
+.endm
+.macro vmulq d,a,b,i
+        mul \d\().4s, \a\().4s, \b\().s[\i]
+.endm
+.macro vmlsq d,a,b,i
+        mls \d\().4s, \a\().4s, \b\().s[\i]
+.endm
+
+.macro mulmodq dst, src, const, idx0, idx1
+        vmulq       \dst,  \src, \const, \idx0
+        vqrdmulhq   \src,  \src, \const, \idx1
+        vmlsq        \dst,  \src, consts, 0
+.endm
+
+.macro mulmod dst, src, const, const_twisted
+        mul        \dst\().4s,  \src\().4s, \const\().4s
+        vqrdmulh   \src,  \src, \const_twisted
+        vmlsq       \dst,  \src, consts, 0
+.endm
+
+.macro montg_reduce a
+        srshr tmp.4S,  \a\().4S, #23
```

Comment:
Is this really a Montgomery reduction? It looks more like Barrett, leveraging somehow that `q` is very close to `2^23`.

What exactly is happening here, and what are the bounds guarantees?

I remember we talked about this at some point, but I cannot remember.

Reply:
To me, this appears to be a shortened version of `reduce32` from the reference implementation. As mentioned above, I followed the current state in PQClean. 
Anyhow, the naming is off.

Reply:
Could you adjust the macro names?